### PR TITLE
various minor tweaks/changes to aws stuff

### PR DESCRIPTION
**Pull request by gene-redpanda**
Added tag merging to support custom tags alongside our existing tags, fixed various formatting issues, cleaned up some deprecated formatting.

Added the GCP splat fixes in too since I missed them before.

### CODE REVIEW

File: aws/vars.tf [hunk 1]
```diff
@@ -194,3 +199,15 @@ variable "tiered_storage_enabled" {
   type        = bool
   default     = false
 }
+
+variable "private_key_path" {
+  type        = string
+  description = "The contents of an SSH key to use for the connection. These can be loaded from a file on disk using the file function. This takes preference over password if provided."
+}
```
Comment (hcoyote) [inline aws/vars.tf:206]: Is this going to force people to set a private_key_path or will the lack of a default here allow this to fall through to the running ssh agent? I guess right now it's implicitly looking for one directly and we're just not setting the variable in a place that people can override easily?

File: aws/vars.tf [hunk 2]
```diff
@@ -143,29 +143,34 @@ variable "public_key_path" {
...
-    }
+  most_recent = true
+
+  filter {
+    name   = "name"
+    values = [
+      "ubuntu/images/hvm-ssd/ubuntu-*-amd64-server-*", "Fedora-Cloud-Base-*.x86_64-hvm-us-west-2-gp2-0",
+      "debian-*-amd64-*", "debian-*-hvm-x86_64-gp2-*'", "amzn2-ami-hvm-2.0.*-x86_64-gp2",
+      "RHEL*HVM-*-x86_64*Hourly2-GP2"
+    ]
```
Comment (hcoyote) [inline aws/vars.tf:154]: For readability, can we do the image name patterns as 1 per line?

**Review by hcoyote — CHANGES_REQUESTED** [review]

File: aws/vars.tf (hunk 1 above)
Comment (gene-redpanda) [reply aws/vars.tf:206]: Well as far as I can tell we just didn't have a variable for private key. So it threw an error on plan. 

I'm fine making that element optional, but I'm pretty sure it's needed for the ansible connection.

File: aws/vars.tf (hunk 2 above)
Comment (gene-redpanda) [reply aws/vars.tf:154]: Sure, nbd

File: aws/vars.tf (hunk 1 above)
Comment (hcoyote) [reply aws/vars.tf:206]: We only need to load the public key to make ansible work. I'm only loading a pub key on my builds and relying upon an agent.

Comment (gene-redpanda) [reply aws/vars.tf]: defaults to null

**Review by hcoyote — APPROVED** [review]

**Pull request merged**